File: examples/python/guides/integrations/scraper_firecrawl.py
# Third-party integration example - requires: pip install firecrawl-py
# See: /guides/web-scraping

import os
from firecrawl import FirecrawlApp

firecrawl = FirecrawlApp(api_key=os.environ["FIRECRAWL_API_KEY"])


# > Firecrawl usage
def scrape_url(url: str) -> dict:
    result = firecrawl.scrape_url(url, params={"formats": ["markdown"]})
    return {
        "url": url,
        "content": result["markdown"],
        "metadata": result.get("metadata", {}),
    }

File: examples/typescript/guides/integrations/scraper-firecrawl.ts
// Third-party integration - requires: pnpm add @mendable/firecrawl-js
// See: /guides/web-scraping

import FirecrawlApp from '@mendable/firecrawl-js';

const firecrawl = new FirecrawlApp({ apiKey: process.env.FIRECRAWL_API_KEY! });

// > Firecrawl usage
export async function scrapeUrl(url: string) {
  const result = await firecrawl.scrapeUrl(url, { formats: ['markdown'] });
  return {
    url,
    content: result.markdown,
    metadata: result.metadata,
  };
}
